### PR TITLE
Add quotes as an include filter to Person api

### LINKED ISSUE #27: Implement a serializer for Person
# Overview
We're currently doing a simple serialization on models by constructing a dict of the fields and their values. Implement a more robust serialization process where the order of the keys can be controlled and one that supports `include` filters.

### PR DESCRIPTION
**Pull request by bellkelly**
# Overview

Add the ability to optionally include the `quotes` field for the `/people` endpoint

# Tasks
- [x] Improve ordering of API response (was completed in #43 )
- [x] Add mixin to support include filters
- [x] Add `include` query param for `quotes`
- [x] Add tests for `include` query param
- [x] Update API docs

I also included some clean up in this PR, like adding docstrings and making use of the `prepared_user` in tests.

# References

resolves #27

### CODE REVIEW

File: docs/api.yaml [hunk 1]
```diff
@@ -60,13 +71,22 @@ paths:
...
+          description: Used to include the optional field `quotes`
+          example: "include=quotes"
       responses:
         "200":
           description: "Person matching criteria:"
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Person"
+                oneOf:
```
Comment (bellkelly) [inline docs/api.yaml:87]: The `oneOf` key works somewhat well for list endpoints, but it doesn't seem to work so well for single items.

There's an FAQ at the bottom of this page in the docs asking about mapping request params to responses, and it doesn't seem to be possible at the moment 😢 https://swagger.io/docs/specification/describing-responses/

File: docs/api.yaml [hunk 2]
```diff
@@ -98,7 +134,8 @@ components:
         quotes:
           type: array
           items:
-            $ref: '#/components/schemas/Quote'
```
Comment (revangel) [inline docs/api.yaml:101]: Did we not want to have the other data in a `Quote` returned?

File: docs/api.yaml (hunk 1 above)
Comment (revangel) [reply docs/api.yaml:87]: That seems ok? If it's a single item, wouldn't we just use what we were doing before?

**Comment by revangel** [comment]
Also, does this implement https://github.com/revangel/nostalgiabot2/issues/12 as well?

**Comment by revangel** [comment]
I was working on the service and ran into [this](https://flask-restful.readthedocs.io/en/latest/reqparse.html):

> The whole request parser part of Flask-RESTful is slated for removal and will be replaced by documentation on how to integrate with other packages that do the input/output stuff better (such as marshmallow). This means that it will be maintained until 2.0 but consider it deprecated. Don’t worry, if you have code using that now and wish to continue doing so, it’s not going to go away any time too soon.

Maybe it's kinda nice that it's going away since I'm a bit confused about what the struct is and how to parse `parsed_args` (ironically)

File: docs/api.yaml (hunk 2 above)
Comment (bellkelly) [reply docs/api.yaml:101]: Do you think we should have more stuff? Previously it was actually just a list of strings so I was fixing this. But I'm open to including more details if you think it's worthwhile.

File: docs/api.yaml (hunk 1 above)
Comment (bellkelly) [reply docs/api.yaml:87]: Yeah, we could just have it show the Person details without the quotes. We just then don't have an example for when choosing to pass the param `include=quotes`

**Comment by bellkelly** [comment]
Nice catch on the deprecation of parsed args. How do you think we should proceed? Should we keep things as they are now, but create a new ticket to change them to something else? Or change them up before merging this flask-restful stuff?

If you want to keep it for now and fix it later, we can chat about the confusing stuff so that we can at least work with it while it's in this state.

File: docs/api.yaml (hunk 2 above)
Comment (revangel) [reply docs/api.yaml:101]: I was just thinking if we would need to expand it in the future if a `Quote` holds more than just content. But I think this is fine for now and we'll just update this when we need to

**Comment by revangel** [comment]
> Nice catch on the deprecation of parsed args. How do you think we should proceed? Should we keep things as they are now, but create a new ticket to change them to something else? Or change them up before merging this flask-restful stuff?
> 
> If you want to keep it for now and fix it later, we can chat about the confusing stuff so that we can at least work with it while it's in this state.

So in the service tickets, I wasn't really using `parsed_args` as is - I made DTOs to have some way of standardizing the inputs to the service calls. I'm not really too partial to `parsed_args` since I destructure them anyway. If you think there's more we can do with `parsed_args` then I don't mind keeping them in as well. Otherwise, the only thing I'm looking for for the service is to have some way to get the API data into the DTOs.

**Comment by bellkelly** [comment]
> Also, does this implement #12 as well?

I missed this comment on my first pass earlier today. It does not implement #12. It wouldn't be too hard to add that in, but maybe that can be done in a separate PR anyway

**Comment by bellkelly** [comment]
> So in the service tickets, I wasn't really using `parsed_args` as is - I made DTOs to have some way of standardizing the inputs to the service calls. I'm not really too partial to `parsed_args` since I destructure them anyway. If you think there's more we can do with `parsed_args` then I don't mind keeping them in as well. Otherwise, the only thing I'm looking for for the service is to have some way to get the API data into the DTOs.

Hm, so maybe if you're ok with this, we keep things as they are now, but come back and see how we can improve them later?

**Comment by revangel** [comment]
> > Also, does this implement #12 as well?
> 
> I missed this comment on my first pass earlier today. It does not implement #12. It wouldn't be too hard to add that in, but maybe that can be done in a separate PR anyway

Yes, another ticket then!

**Comment by revangel** [comment]
> > So in the service tickets, I wasn't really using `parsed_args` as is - I made DTOs to have some way of standardizing the inputs to the service calls. I'm not really too partial to `parsed_args` since I destructure them anyway. If you think there's more we can do with `parsed_args` then I don't mind keeping them in as well. Otherwise, the only thing I'm looking for for the service is to have some way to get the API data into the DTOs.
> 
> Hm, so maybe if you're ok with this, we keep things as they are now, but come back and see how we can improve them later?

My philosophy for this project is to go fast and hard on the deliverables and refine later since we have the advantage of being able to experiment freely - so yes!

**Pull request merged**